### DISCUSSION TITLE
Supporting Subseconds

### DISCUSSION THREAD

**Discussion by fswetnam** [wojtekmaj/react-datetime-picker#427] [Ideas]
Is there any way this library can be updated to support subseconds? (milliseconds up to 3 decimals) My project requires it, and we are currently utilizing your library for datetime selection, but as it doesn't support subseconds we are probably going to have to do a refactor of our current code.

Thanks!

**Comment by DeeJaayArgh** [comment]
I'm also looking to do this and have the format be an ISO string.  with DateTimePicker I was able to get the format the way I wanted but couldn't get the millisecond control that one can get with a datetime-local native input.

DateTimePicker
```
      <DateTimePicker
        onChange={onChange}
        value={value}
        step={0.001}
        format="yyyy-MM-ddTHH:mm:ssZ"
        disableClock={true}
      />
```
<img width="270" height="60" alt="image" src="https://github.com/user-attachments/assets/5ba5f261-e921-46e1-870c-9f220faeef9c" />

input with type datetime-local
```
     <input
        onChange={(e) => handleDatetime(e)}
        type="datetime-local"
        value={datetime}
        step="0.001"
      />
```
<img width="292" height="41" alt="image(1)" src="https://github.com/user-attachments/assets/38e9e2f9-094e-4ab0-99d5-6cf63329cd46" />
